arch/eventloop_posix_tcp.c

```diff
@@ -490,6 +490,70 @@ TCP_registerListenSocketDomainName(UA_ConnectionManager *cm, const char *hostnam
     return UA_STATUSCODE_GOOD;
 }
 
+#ifdef _WIN32
+static UA_RegisteredFD *
+findRegisteredFD(TCPConnectionManager *cm, uintptr_t connectionId) {
+    UA_RegisteredFD *rfd;
+    LIST_FOREACH(rfd, &cm->fds, es_pointers) {
+        if (rfd->fd == connectionId) {
+            return rfd;
+        }
+    }
+    return NULL;
+}
+
+static UA_DelayedCallback
+*createExplicitlyDelayedShutdownCallback(UA_ConnectionManager *cm, UA_RegisteredFD *rfd, UA_Callback callback) {
+    UA_DelayedCallback *explicitlyDelayedShutdownCallback = (UA_DelayedCallback*) UA_malloc(sizeof(UA_DelayedCallback));
+    if (!explicitlyDelayedShutdownCallback) {
+        return NULL;
+    }
+    explicitlyDelayedShutdownCallback->callback = callback;
+    explicitlyDelayedShutdownCallback->application = cm;
+    UA_FD *fd = (UA_FD *) UA_malloc(sizeof(UA_FD));
```

Comment:
The FD always fits into the size of a pointer.
So you don't have to allocate this memory here.
Just cast the pointer itself.